src/bin2pat/bin2pat.cpp

```diff
@@ -81,6 +83,37 @@ void processArgs(
 				return;
 			}
 		}
+		else if (args[i] == "-l" || args[i] == "--list") {
+			// Ensure -l --list is not the last thing in args
+			if (&args[i] == &args.back()) {
+				printErrorAndDie("input file missing");
+				return;
+			}
+
+			std::ifstream inputObjects(args[++i]);
+			std::string object;
+
+			if (inputObjects.is_open()) {
+				// Read LIST_FILE until EOF
+				while (inputObjects >> object) {
```

Comment:
[`std::istream::operator>>()`](https://en.cppreference.com/w/cpp/io/basic_istream/operator_gtgt) stops reading after encountering the first whitespace character. Therefore, if a file path contains spaces, it will be split into multiple paths. This is rare, but might happen as some users/applications put spaces into file names. Actually, [one of our tests failed](https://retdec-tc.avast.com/viewLog.html?buildId=1183&buildTypeId=Retdec_LinuxBuild&tab=buildLog&_focus=3450) exactly because of this issue.

To illustrate, an input file containing a single line
```
my file with spaces.txt
```
would be parsed into the following objects:
```
my
file
with
spaces.txt
```

Therefore, to be safe, we should switch to [`std::getline()`](https://en.cppreference.com/w/cpp/string/basic_string/getline) to read the objects from the input file. This can be simply done by replacing
```cpp
while (inputObjects >> object) {
```
with
```cpp
while (std::getline(inputObjects, object)) {
```

Reply:
I was mistakenly under the impression that the delimiter for reading a string from a stream was a new line character. I will resolve this with the above suggestion.